e2e/namespaces/namespaces_test.go

```diff
@@ -0,0 +1,154 @@
+	namespaceIDs = append(namespaceIDs, "NamespaceB")
+
+	run := func(jobspec, ns string) string {
+		jobID := "test-namespace-" + uuid.Generate()[0:8]
+		must.NoError(t, e2eutil.Register(jobID, jobspec))
+		namespacedJobIDs = append(namespacedJobIDs, [2]string{ns, jobID})
+		expected := []string{"running"}
+		require.NoError(t, e2eutil.WaitForAllocStatusExpected(jobID, ns, expected), "job should be running")
+		return jobID
+	}
+
+	jobA := run("./input/namespace_a.nomad", "NamespaceA")
+	jobB := run("./input/namespace_b.nomad", "NamespaceB")
+	jobDefault := run("./input/namespace_default.nomad", "")
+
+	// exercise 'nomad job status' filtering
+	parse := func(out string) []map[string]string {
+		rows, err := e2eutil.ParseColumns(out)
+		require.NoError(t, err, "failed to parse job status output: %v", out)
+
+		result := make([]map[string]string, 0, len(rows))
+		for _, row := range rows {
+			jobID := row["Job ID"]
+			if jobID == "" {
+				jobID = row["ID"]
+			}
+			if strings.HasPrefix(jobID, "test-namespace-") {
+				result = append(result, row)
+			}
+		}
+		return result
+	}
+
+	out, err := e2eutil.Command("nomad", "job", "status", "-namespace", "NamespaceA")
+	require.NoError(t, err, "'nomad job status -namespace NamespaceA' failed")
+	rows := parse(out)
+	must.Len(t, 1, rows)
+	must.Eq(t, rows[0]["ID"], jobA)
+
+	out, err = e2eutil.Command("nomad", "job", "status", "-namespace", "NamespaceB")
+	require.NoError(t, err, "'nomad job status -namespace NamespaceB' failed")
+	rows = parse(out)
+	must.Len(t, 1, rows)
+	must.Eq(t, rows[0]["ID"], jobB)
+
+	out, err = e2eutil.Command("nomad", "job", "status", "-namespace", "*")
+	require.NoError(t, err, "'nomad job status -namespace *' failed")
+	rows = parse(out)
+	must.Len(t, 3, rows)
+
+	out, err = e2eutil.Command("nomad", "job", "status")
+	require.NoError(t, err, "'nomad job status' failed")
+	rows = parse(out)
+	must.Len(t, 1, rows)
+	must.Eq(t, rows[0]["ID"], jobDefault)
+
+	// exercise 'nomad status' filtering
+
+	out, err = e2eutil.Command("nomad", "status", "-namespace", "NamespaceA")
+	require.NoError(t, err, "'nomad job status -namespace NamespaceA' failed")
+	rows = parse(out)
+	must.Len(t, 1, rows)
+	must.Eq(t, rows[0]["ID"], jobA)
+
+	out, err = e2eutil.Command("nomad", "status", "-namespace", "NamespaceB")
+	require.NoError(t, err, "'nomad job status -namespace NamespaceB' failed")
+	rows = parse(out)
+	must.Len(t, 1, rows)
+	must.Eq(t, rows[0]["ID"], jobB)
+
+	out, err = e2eutil.Command("nomad", "status", "-namespace", "*")
+	require.NoError(t, err, "'nomad job status -namespace *' failed")
+	rows = parse(out)
+	must.Len(t, 3, rows)
+
+	out, err = e2eutil.Command("nomad", "status")
+	require.NoError(t, err, "'nomad status' failed")
+	rows = parse(out)
+	must.Len(t, 1, rows)
+	must.Eq(t, rows[0]["ID"], jobDefault)
+
+	// exercise 'nomad deployment list' filtering
+	// note: '-namespace *' is only supported for job and alloc subcommands
```

Comment:
Not related to this PR, but I think we have `nomad deployment list -namespace=*` now 🎉